### DISCUSSION TITLE
Manually define dependencies of a target

### DISCUSSION THREAD

**Discussion by MxNl** [ropensci/targets#828] [Help] [answered]
I have a pipeline where one target (summary) uses two upstream targets as input. But I only want to rerun the target if the target rerun_summary has changed to run it every month (or minute as inthe example). Maybe I am not seeing the most obvious or simple solution :-). I thought of using something like `tar_change()` and use the target rerun_summary for the change argument, but this function has no pattern argument.

Example:
``` r
library(targets)
tar_script({
    library(lubridate)
    library(dplyr)
    list(tar_target(data, tar_group(group_by(data.frame(x = sample.int(10), 
        groups = rep_len(c("a", "b"), length.out = 10)), groups)), 
        iteration = "list"), tar_target(rerun_summary, lubridate::minute(Sys.time()), 
        cue = tar_cue("always")), tar_target(summary, mean(data$x, 
        na.rm = TRUE), pattern = data))
})
tar_make()
#> 
#> Attache Paket: 'lubridate'
#> 
#> Die folgenden Objekte sind maskiert von 'package:base':
#> 
#>     date, intersect, setdiff, union
#> 
#> 
#> Attache Paket: 'dplyr'
#> 
#> Die folgenden Objekte sind maskiert von 'package:stats':
#> 
#>     filter, lag
#> 
#> Die folgenden Objekte sind maskiert von 'package:base':
#> 
#>     intersect, setdiff, setequal, union
#> 
#> * start target data
#> * built target data
#> * start target rerun_summary
#> * built target rerun_summary
#> * start branch summary_b74354e1
#> x error branch summary_b74354e1
#> * end pipeline
#> Error : $ operator is invalid for atomic vectors
#> Error: callr subprocess failed: $ operator is invalid for atomic vectors
#> Visit https://books.ropensci.org/targets/debugging.html for debugging advice.
```

<sup>Created on 2022-04-19 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

I really appreciate any hint. Thanks, Max

**Comment by wlandau** [comment, marked as answer]
You could define a target between `data` and `summary` that uses `cue = tar_cue(mode = "never")`. That way, the intermediate target does not change even `data` changes, so `summary` will not rerun. Sketch:

```r
# _targets.R file
library(targets)
list(
  tar_target(data, ...),
  tar_target(data_constant, data, cue = tar_cue(mode = "never")),
  tar_target(rerun_summary, ...),
  tar_target(
    summary, {
      rerun_summary
      mean(data_constant$x, na.rm = TRUE)
    },
    pattern = data_constant
  )
)
```

**Reply by MxNl** [reply]
@wlandau: Thanks a lot for the quick response. Because of your correct answer to the descibed task, I realised, that I forgot to mention a detail and therefore my question is just a little bit different. This is why I marked your reply as solution and raised a new question [here](https://github.com/ropensci/targets/discussions/830)
